src/dos/drive_local.cpp

```diff
@@ -375,21 +390,75 @@ bool localDrive::FindNext(DOS_DTA & dta) {
 	return true;
 }
 
-bool localDrive::GetFileAttr(char * name,Bit16u * attr) {
+bool localDrive::GetFileAttr(char *name, uint16_t *attr)
+{
 	char newname[CROSS_LEN];
 	safe_strcpy(newname, basedir);
 	safe_strcat(newname, name);
 	CROSS_FILENAME(newname);
 	dirCache.ExpandName(newname);
 
+#if defined(WIN32)
+	Bitu attribs = GetFileAttributes(newname);
+	if (attribs == INVALID_FILE_ATTRIBUTES) {
+		DOS_SetError((uint16_t)GetLastError());
+		return false;
+	}
+	*attr = attribs & 0x3f;
+	return true;
+#else
+	struct stat status;
+	if (stat(newname, &status) == 0) {
+		*attr = status.st_mode & S_IFDIR ? 0 : DOS_ATTR_ARCHIVE;
+		if (status.st_mode & S_IFDIR)
+			*attr |= DOS_ATTR_DIRECTORY;
```

Comment:
For the OS-specific branches that use C-style WIN32 vs. Posix calls, checkout: 

https://en.cppreference.com/w/cpp/filesystem

This gives us OS-agnostic operations (including permissions, and so on).

``` c++
#include "std_filesystem.h"

// then use.. std_fs:: to get the std::filesystem:: namespace:
std_fs::path
std_fs::exists
...